app/stac_api/serializers.py

```diff
@@ -521,13 +519,16 @@ def update_or_create(self, look_up, validated_data):
         return asset, created
 
     def validate(self, attrs):
-        if not self.partial:
-            validate_asset_name_with_media_type(attrs.get('name'), attrs.get('media_type'))
-        if self.partial and ('name' in attrs or 'media_type' in attrs):
-            validate_asset_name_with_media_type(
-                attrs.get('name', self.instance.name),
-                attrs.get('media_type', self.instance.media_type)
+        name = attrs['name'] if not self.partial else attrs.get('name', self.instance.name)
+        if self.partial and 'media_type' not in attrs:
+            media_type = validate_and_get_media_type(name, self.instance.media_type)
+            validate_asset_name_with_media_type(name, media_type)
+        else:
+            media_type = validate_and_get_media_type(
+                name, attrs['media_type'], media_type_str_is_normalized=False
             )
+            validate_asset_name_with_media_type(name, media_type)
+            attrs['media_type'] = media_type.media_type_str
```

Comment:
I'm not sure it is a good idea to overwrite the media type for the normalization here in the validate method. Is it not possible to do this in the field validation ?

Reply:
After looking again at the django rest framework docu, I think the code could be simplified and made more understandable by using a `CustomField` https://www.django-rest-framework.org/api-guide/fields/#custom-fields. The `to_representation()` method would return the normalized media type while the `to_internal()` method would parse the media type string and raise a ValidationError in case of invalid string and return a `MediaType` object. I would also create a `MediaType` class instead of a named tuple, the class would then have a `__str__()` method that would normalize the media type. The `get_media_type_comparable()` function should then be rename `parse_media_type()` and would return a MediaType object instance.
In theorie in the model we don't need any media type validation as they are coming either from the serializer or from the admin interface choice field, but I would keep/add there a very simple quick validation without normalization (a validate function that checks that the string is in the media type dictionary. The media type dictionary would use the normalize media type as key.